cumulusci/core/keychain/EncryptedFileProjectKeychain.py

```diff
@@ -15,7 +16,7 @@ def config_local_dir(self):
         except AttributeError:
             # Handle a global config passed as project config
             config_local_dir = self.project_config.config_local_dir
-        return os.path.join(os.path.expanduser("~"), config_local_dir)
+        return str(Path(Path.home(), config_local_dir))
```

Comment:
I think we want to stop using BaseGlobalConfig.config_local_dir altogether; it's not very useful.

This property should become:

    try:
        config_local_dir = self.project_config.global_config_obj.cumulusci_config_dir
    except AttributeError:
        # Handle a global config passed as a project config
        config_local_dir = self.project_config.cumulusci_config_dir
    return config_local_dir

or else add a cumulusci_config_dir property to BaseProjectConfig as well (delegating to its global_config_obj) so that we don't need to do the try/except here.

Reply:
I was afraid to remove the os.path.join because I wasn't sure whether it would sometimes be invoked like this:

```
>>> os.path.join("/foo/bar", "/a/b")
'/a/b'
```

Reply:
@prescod I'm not following. The point of this property is to get the cumulusci config dir from the GlobalConfig object.

Reply:
I honestly have no idea what the two branches of this function are intended to do, so I'm basically reluctant to touch it.

Why would `self.project_config.global_config_obj.config_local_dir` be sometimes missing and why would it be available at `self.project_config.config_local_dir`. If its supposed to be an override passed as a "project config" (as opposed to a global config) then why not look in the project first?

Everything about how these config objects interact makes me nervous, so I am reluctant to change them more than necessary.

Reply:
@prescod `BaseProjectKeychain.__init__` takes an arg called `project_config`, but in reality it is sometimes passed an instance of BaseProjectConfig (when running inside a project directory) and sometimes an instance of BaseGlobalConfig (when not inside a project directory, because keychain entries like orgs and services can also be configured globally). So the line inside the `try` is for when we have a project config, and the line in the except is for when we actually have a global config.

Reply:
@prescod let me know if that helped. If not, maybe we should discuss in real time.